### DISCUSSION TITLE
Translatable automatic title formats for multi-sites

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#15494] [Ideas] [answered]
Entry titles can either use a pre-defined format or you can show the title field so authors can manually set a title. With the latter option, you can also use translation methods to set your title to be translatable or not. But you _can't_ do that with automatic titles.

In some cases, we want to use an automatic title format, but have a different format for different sites/languages. I can see two ways to support this:

- Allow using translation methods in conjunction with automatic title formats. The format can then use translations, or output different values based on the entry's site, and the format would either propagate or not based on the selected translation method.
- The entry type config should show a separate title format field for each site, similar to settings for section URLs and template paths.

**Comment by brandonkelly** [comment, marked as answer]
The Title Format is a Twig template, so you can do whatever you want in there—even include an external template from your `templates/` folder with an `{% include %}` tag, and put as much logic in there as you need. The entry will be available to it via an `object` variable.

```twig
{% if object.site.language == 'de' %}
  ...
{% else %}
  ...
{% endif %}
```

**Reply by MoritzLost** [reply]
@brandonkelly Ah I didn't realize the title format is evaluated for each site separately, that makes sense, thanks!
